### PR TITLE
Initial work on setting rosout QoS

### PR DESCRIPTION
**Pull request by J0las**
This is the first implementation of explicitly setting the rosout QoS.
I am sure there will be plenty of debate around what QoS settings to use. Think of them as placeholders.
We could also consider moving the constructor implementation into the cpp file.
This way a change in the implementation only means rebuilding the library and not all the other cpp files that include the header as well.
This is also of interest for the telemetry team @maf0115 & @di-math please submit your feedback.

### CODE REVIEW

**Review by Philipp66904 — DISMISSED** [review]

**Comment by maf0115** [comment]
Is there a reason behind setting the queue size specifically to 1?

**Comment by maf0115** [comment]
Other than that, the other measures you took for setting the QoS look good to me

**Comment by J0las** [comment]
> Is there a reason behind setting the queue size specifically to 1?

To keep the memory footprint as low as possible.

**Comment by J0las** [comment]
> Other than that, the other measures you took for setting the QoS look good to me

Please also check if this is still good enough with a Wifi connection. I only checked if it compiles.

**Comment by di-math** [comment]
@J0las - looking good to me

**Review by J0hn-Lawrence — COMMENTED** [review]
Are you planning to test this with WiFi on HW before merging @J0las  @di-math @maf0115 ?

**Comment by di-math** [comment]
> Are you planning to test this with WiFi on HW before merging @Jolas @di-math @maf0115 ?

Ich glaube du hast irgendein Random Spanier markiert lol

**Comment by di-math** [comment]
> Are you planning to test this with WiFi on HW before merging @Jolas @di-math @maf0115 ?

@J0hn-Lawrence We certainly should - will see to have @maf0115 finish his work very soon.

**Comment by J0las** [comment]
> > Are you planning to test this with WiFi on HW before merging @jolas @di-math @maf0115 ?
> 
> Ich glaube du hast irgendein Random Spanier markiert lol

Lol

**Comment by J0hn-Lawrence** [comment]
> > Are you planning to test this with WiFi on HW before merging @jolas @di-math @maf0115 ?
> 
> Ich glaube du hast irgendein Random Spanier markiert lol

Haha, thanks for letting me know :laughing:

File: include/common_package/common_node.hpp [hunk 1]
```diff
@@ -12,14 +12,45 @@
...
 #include "interfaces/msg/job_finished.hpp"
 
 using namespace std::chrono_literals;
 
 namespace common_lib {
+
+namespace {
+
+/**
+ * @brief rosout QoS to explicitly set the it
```
Comment (Philipp66904) [inline include/common_package/common_node.hpp]: What do you want to say here?

**Review by Philipp66904 — CHANGES_REQUESTED** [review]

File: include/common_package/common_node.hpp (hunk 1 above)
Comment (J0las) [reply include/common_package/common_node.hpp]:
```suggestion
 * @brief rosout QoS to explicitly set the rosout for  Nodes subclassing CommonNode
```

Better?

Comment (Philipp66904) [reply include/common_package/common_node.hpp]: Yes, now it makes sense for me

Comment (Philipp66904) [reply include/common_package/common_node.hpp]: Autoformatter doesn't like this chance 😂
@J0las

Comment (J0las) [reply include/common_package/common_node.hpp]: Mimimimi

**Comment by Philipp66904** [comment]
Can you add the same chances to common_package_py, too?
@J0las

**Review by Philipp66904 — APPROVED** [review]

**Review by J0hn-Lawrence — APPROVED** [review]

**Pull request merged**